src/main/proto/netflix/titus/titus_base.proto

```diff
@@ -301,6 +301,14 @@ message SignedAddressAllocation {
     /// (Required) The IP address and metadata that was allocated.
     AddressAllocation addressAllocation = 1;
 
-    // (Required) This is an x509v3 signed version of the address allocation.
-    bytes signedAddressAllocation = 2;
-};
\ No newline at end of file
+    // authoritative publicKey is the key that came from the database
+    bytes authoritativePublicKey = 3;
```

Comment:
Why do we pass the public key in the message? Doesn't that make this vulnerable to MITM?

Reply:
We pass the authoritative public key so it can be looked up. Without it, we have to take all the public keys we trust and iterate over them to see if any of them are valid.

Reply:
Seems like we should have a better way to identify the public key to use without just passing the key itself. Otherwise, a MITM could sign the host public key and then pass their own public key to use for verification.

Reply:
@andrew-leung So, there's a database (PG) that stores all the canonical versions of the trusted keys. You can check against that to validate that the public key is not part of MITM.

Reply:
Do keys have a unique ID? Would it be better to pass that than the key itself?

Reply:
We could pass a sha512 hash of the key instead.

Reply:
but the entire idea of the public key is that then the user doesn't need to go and iterate over all the keys to figure out which key was used to sign the allocation.

Reply:
I was thinking when the signing service signs the `hostPublicKey` it would know it used "I used key ID XYZ to sign". When the Agent or VPC Service verifies the signature (I'm assuming it one of those services doing it), it would resolve XYZ to a public key from the PG DB mentioned above. Seems like it needs to call the PG DB anyways to validate that the provided key exists in the DB.

Reply:
If there is no ID for the key, then it's not a problem to just pass the whole key.